engine/runtime/src/main/java/org/enso/interpreter/runtime/data/EnsoDateTime.java

```diff
@@ -138,6 +144,24 @@ public EnsoTimeZone zone() {
     return new EnsoTimeZone(dateTime.getZone());
   }
 
+  @Builtin.Method(name = "plus_builtin", description = "Adds a duration to this date time")
+  @Builtin.Specialize
+  @Builtin.WrapException(from = UnsupportedMessageException.class, to = PanicException.class)
+  @CompilerDirectives.TruffleBoundary
+  public EnsoDateTime plus(Object durationObject, InteropLibrary interop)
+      throws UnsupportedMessageException {
+    return new EnsoDateTime(dateTime.plus(interop.asDuration(durationObject)));
+  }
```

Comment:
Normally if a type mismatch happens, we will throw a `Type_Error` or `Unsupported_Argument_Types`.

If I understand this code correctly, here instead we will throw `UnsupportedMessageException`. I think maybe we should wrap it into some Enso-specific type? I think we could re-use one of the two above or alternatively create a new one. I'm just not convinced we should expose a Truffle exception type to our users - instead we should have an exception type that shows up somewhere in Enso like the two above (i.e. it has its `type Type_Error` definition with documentation, `to_display_text` etc.)